help/share-and-upload-files.md

```diff
@@ -1,21 +1,58 @@
 # Share and upload files
 
-Attach files to messages, including images, documents, sound, and video.
+Zulip supports attaching multiple files to messages, including images,
+documents, sound, and video. You can edit the names of the files others see
+after you upload them.
 
-First, [open the compose box](/help/open-the-compose-box). Then
+Zulip will automatically generate a **thumbnail** for each file when you send
+the message, if it can. Image thumbnails will be shown directly in the message,
+and you can click on a thumbnail to [view the full image](/help/view-and-browse-images).
 
-* **Drag and drop** files into the compose box.
-* **Copy and paste** files into the compose box.
-* Click the **paperclip** (<i class="fa fa-paperclip"></i>) icon at
-  the bottom of the compose box to find files on your computer.
+## Uploading files
 
-Zulip will insert a link to the file, in Markdown format:
-`[link text](URL)`. You can modify the `link text` to whatever you want.
+{start_tabs}
 
-Zulip will automatically generate a **thumbnail** of the file when you send
-it, if it can. You can
-[preview the message](/help/preview-your-message-before-sending) before
-sending to see what the thumbnail will look like.
+{tab|via-markdown}
+
+{!start-composing.md!}
+
+1. Drag and drop files, or copy and paste one or more files into the compose
+   box. Zulip will upload the files, and insert named links using
+   [Markdown formatting](/help/format-your-message-using-markdown#links):
+   `[File name](URL)`.
+
+1. _(optional)_ Modify the file name as desired.
```

Comment:
I think "file name" isn't a good way to describe this -- if a user downloads the file, for example, the file name they get is basically the end of the URL, and they can't change the URL without re-uploading the file. 

I think maybe "Link label", which defaults to the file name, would be a good description.

Reply:
Mm, OK. So maybe call it "Link label", and add a sentence that says:

> The link label will default to the name of the uploaded file.

And then the following instruction will refer to the "link label" as well.

Reply:
In that case I think we should also change https://zulip.com/help/insert-a-link#insert-a-link_1 to refer to "link label" rather than "link text". I don't love the current wording there, so maybe something like this?

> To create a named link, use `[ ]` around the link label, and `( )` around the URL: `[Link label](URL)`.

Reply:
Or should it be "Link name", to align with "Link name"?

Reply:
I think "Link text" is probably most intuitive -- it's just the text that's going to be linkified with your link. "label"/"name" kinda suggest it has some special meaning, whereas it's really just the text used to link your file.

Reply:
> Mm, OK. So maybe call it "Link label", and add a sentence that says:
> 
> > The link label will default to the name of the uploaded file.

Maybe this could go in a `tip` block after the instructions block?

I'm not sure about "link label", but I think either "link name" or "link text" would work well.

Reply:
Ok, let's give "link name" a try, then.

Reply:
It's not super intuitive with me, but I guess we can use "link text". Let's still tweak the wording on [zulip.com/help/insert-a-link#insert-a-link_1](https://zulip.com/help/insert-a-link#insert-a-link_1), even though we aren't changing the "link text" part.

Reply:
> Maybe this could go in a tip block after the instructions block?

Sure, we can try it that way.